### ISSUE TITLE
Support for asp.net core targeting full framework

### ISSUE THREAD

**Issue by blebaron** [aws/aws-xray-sdk-dotnet#36]
I'm working on an asp.net project that is targeting the full .net framework. I get the following error when attempting to use the asp.net core XRay library:

```
System.MissingMethodException: Method not found: 'Boolean Amazon.XRay.Recorder.Core.AWSXRayRecorder.IsLambda()'.
   at Amazon.XRay.Recorder.Handlers.AspNetCore.Internal.AWSXRayMiddleware.ProcessHTTPRequest(HttpContext context)
   at Amazon.XRay.Recorder.Handlers.AspNetCore.Internal.AWSXRayMiddleware.<Invoke>d__13.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.<Invoke>d__7.MoveNext()
```

I haven't dug in to try and find where the issue is but I have created a simple example that illustrates the issue here: https://github.com/blebaron/AWSXRayExample

I created the example by using the VS 2017 asp.net core api new project template and then modifying it to target the `net461`

**Comment by yogiraj07** [comment]
Hi @blebaron ,

The reason for your exceptions is, you are using `ASP.NETCore 2.0` X-Ray package for `net461` framework.
`IsLambda()` is present only for `netcore2.0` framework. As a result, X-Ray SDK doesn't find this method on `net461` framework.
Reference: https://github.com/aws/aws-xray-sdk-dotnet/blob/master/sdk/src/Handlers/AspNetCore/Internal/AWSXRayMiddleware.cs#L225

I suggest to use ASP.NET X-Ray handler for `net461` framework : https://github.com/aws/aws-xray-sdk-dotnet/tree/master#aspnet-framework-net--nuget

Is there any specific reason for a use case where you change the framework from `netcore2.0` to `net461`?

Thanks,
Yogi

**Comment by blebaron** [comment]
Sorry for the delayed response. I got this working by manually pulling out the .net core assemblies from the nuget packages and referencing them directly and then forgot that I had logged an issue.

I'm using the asp.net core MVC constructs (startup, kestrel, etc...) inside of a full framework project. That's why I can't use the `net461` package. The example I linked to illustrates exactly how I have things set up.
